crdt/g_counter.pony

```diff
@@ -80,7 +94,7 @@ class ref GCounter[A: (Integer[A] val & Unsigned) = U64]
     Accepts and returns a convergent delta-state.
     """
     try
-      let v' = _data.upsert(_id, value', {(v: A, value': A): A => v + value' })?
+      let v' = _data.upsert(_id, value', this~_sum())?
```

Comment:
I think doing it this way will result in an object allocation every time `increment` is called. Since the `_sum` method requires no state from `this`, I think this should be a lambda primitive instead (a lambda that doesn't capture and references from the outer scope).

There are several ways to do this, but I think the best is probably to move the `GCounter[A]._sum` method to a `saturated_sum[A]` method on a primitive named `_Math` in a file called `_math.pony`.

Then here, use the inline lambda syntax as it did before, replacing the `+` with a call to that new primitive method.

This way, no object allocations are required, and LLVM should be able to inline the `addc` logic directly into the anonymous lambda primitive.

Reply:
Ah, interesting. So, creating an inline lambda like: `{(x, y) => x + y }` does not allocate a new anonymous class instane for each call?

Reply:
It only allocates if you have captures. If you have no captures, it will translate to an anonymous primitive instead of a class.